flink-ai-flow/ai_flow/api/ops.py

```diff
@@ -0,0 +1,582 @@
+    elif isinstance(model_info, Text):
+        model_meta = get_ai_flow_client().get_model_by_name(model_info)
+    else:
+        model_meta = get_ai_flow_client().get_model_by_id(model_info)
+
+    if isinstance(model_version_info, ModelVersionMeta) or model_version_info is None:
+        model_version_info = None
+    elif isinstance(model_version_info, Text):
+        model_version_info = get_ai_flow_client().get_model_version_by_version(version=model_version_info,
+                                                                               model_id=model_meta.uuid)
+
+    return user_define_operation(model_info=model_meta,
+                                 executor=executor,
+                                 model_version_info=model_version_info,
+                                 name=name,
+                                 operation_type='push_model',
+                                 **kwargs)
+
+
+def user_define_operation(
+        executor=None,
+        input_data: Union[None, Channel, List[Channel]] = None,
+        output_num=1,
+        name: Text = None,
+        operation_type: Text = 'user_define_operation',
+        **kwargs) -> Union[NoneChannel, Channel, Tuple[Channel]]:
+    """
+    User defined operator.
+
+    :param operation_type: The type of the operation.
+    :param executor: The user defined function in operator. User can write their own logic here.
+    :param input_data: List of input data. It contains multiple channels from the operators which generate data.
+    :param output_num: The output number of the operator. The default value is 1.
+    :param name: Name of this operator.
+    :param kwargs:
+    :return: NoneChannel or Channel or Tuple[Channel].
+    """
+    node = AINode(name=name,
+                  executor=executor,
+                  properties=None,
+                  output_num=output_num,
+                  node_type=operation_type,
+                  **kwargs)
+    add_ai_node_to_graph(node, inputs=input_data)
+    outputs = node.outputs()
+    if 0 == output_num:
+        output: NoneChannel = outputs[0]
+        return output
+    else:
+        if 1 == len(outputs):
+            output: Channel = outputs[0]
+            return output
+        else:
+            output: List[Channel] = []
+            for i in outputs:
+                tmp: Channel = i
+                output.append(tmp)
+            return tuple(output)
+
+
+def action_on_event(job_name: Text,
+                    event_key: Text,
+                    event_value: Text,
+                    event_type: Text = UNDEFINED_EVENT_TYPE,
+                    condition: MetCondition = MetCondition.NECESSARY,
+                    action: TaskAction = TaskAction.START,
+                    life: EventLife = EventLife.ONCE,
+                    value_condition: MetValueCondition = MetValueCondition.EQUAL,
+                    namespace: Text = DEFAULT_NAMESPACE,
+                    sender: Text = None
+                    ):
```

Comment:
The parameter order here seems a little unclear, can we adjust the parameter order a little bit to put the `condition`, `value_condition` and `action` to the last?